arm-containerregistry/2017-06-01-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,1728 @@
+          "type": "string"
+        }
+      }
+    },
+    "RegistryUpdateParameters": {
+      "description": "The parameters for updating a container registry.",
+      "properties": {
+        "tags": {
+          "description": "The tags for the container registry.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU of the container registry."
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryPropertiesUpdateParameters",
+          "description": "The properties that the container registry will be updated with.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryPropertiesUpdateParameters": {
+      "description": "The parameters for updating the properties of a container registry.",
+      "properties": {
+        "adminUserEnabled": {
+          "description": "The value that indicates whether the admin user is enabled.",
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountParameters",
+          "description": "The parameters of a storage account for the container registry. Only applicable for Basic SKU. If specified, the storage account must be in the same physical location as the container registry."
+        }
+      }
+    },
+    "RegistryListResult": {
+      "description": "The result of a request to list container registries.",
+      "properties": {
+        "value": {
+          "description": "The list of container registries. Since this list may be incomplete, the nextLink field should be used to request the next list of container registries.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Registry"
+          }
+        },
+        "nextLink": {
+          "description": "The URI that can be used to request the next list of container registries.",
+          "type": "string"
+        }
+      }
+    },
+    "RegistryListCredentialsResult": {
+      "description": "The response from the ListCredentials operation.",
+      "properties": {
+        "username": {
+          "description": "The username for a container registry.",
+          "type": "string"
+        },
+        "passwords": {
+          "description": "The list of passwords for a container registry.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RegistryPassword"
+          }
+        }
+      }
+    },
+    "RegistryPassword": {
+      "description": "The login password for the container registry.",
+      "properties": {
+        "name": {
+          "description": "The password name.",
+          "enum": [
+            "password",
+            "password2"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PasswordName",
+            "modelAsString": false
```

Comment:
Setting ```"modelAsString":false``` implies that every time to add or remove an element from this enum, it would be a breaking change. Please ensure you are okay with that.